src/pymorize/global_attributes.py

```diff
@@ -0,0 +1,277 @@
+"""
+Global Attributes
+
+Controlled Vocabularies for CMIP6 define a set of required global attributes that must
+be set on a CMIP6 dataset. Setting some of the global attributes is relies on user input
+and setting others is done automatically by consulting the controlled vocabularies. Also,
+some of the global attributes are retrieved from the CMIP6 Table Header for a given variable.
+
+The required user input for global attributes are set on rule via the yaml configuration
+file.
+
+Use the `get_global_attributes` method on `DataRequestVariable` to get the global attributes.
+"""
+
+import re
+import uuid
+
+import xarray as xr
+
+_parent_fields = (
+    "branch_method",
+    "branch_time_in_child",
+    "branch_time_in_parent",
+    "parent_experiment_id",
+    "parent_mip_era",
+    "parent_source_id",
+    "parent_time_units",
+    "parent_variant_label",
+)
+
+
+class GlobalAttributes:
```

Comment:
Make use of a factory here for `CMIP6GlobalAttributes`. I will take care of that.

Reply:
Done!